clang/lib/AST/ExprConstant.cpp

```diff
@@ -4556,7 +4566,7 @@ static CompleteObject findCompleteObject(EvalInfo &Info, const Expr *E,
         NoteLValueLocation(Info, LVal.Base);
         return CompleteObject();
       }
-    } else {
+    } else if (AK != clang::AK_Dereference) {
```

Comment:
We no longer have any `else`, is that correct?

Reply:
Correct, in the dereference case, the object needs to be valid